test/e2e/ssm/command.go

```diff
@@ -82,3 +88,34 @@ func RunCommand(ctx context.Context, client *ssm.Client, instanceId, command str
 
 	return commandOutput, nil
 }
+
+// there is no built in wait for instance to be registered with ssm
+// WaitForInstance uses DescribeInstanceInformation in a loop to wait for it be registered
+// see: https://github.com/aws/aws-cli/issues/4006
+func WaitForInstance(ctx context.Context, client *ssm.Client, instanceId string, logger logr.Logger) error {
+	waitCtx, cancel := context.WithTimeout(ctx, instanceRegisterTimeout)
```

Comment:
Would it be better to pass this in with the timeout? Another option is to do what the aws internal retry code does and keep track of start and end times and keep looping while timeremaining > 0.

Reply:
I think this is fine for our usecase since we always want to wait the same amount

If we ever want to make the timeout "configurable", we could check if the context passed in has a timeout and if it does, respect it. Else, configure it. This would guarantee that the method doesn't block forever but still allows to override the default timeout.